blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/EnumReader.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2016 Rackspace.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.io;
+
+import com.google.common.collect.Table;
+import com.rackspacecloud.blueflood.outputs.formats.MetricData;
+import com.rackspacecloud.blueflood.rollup.Granularity;
+import com.rackspacecloud.blueflood.types.BluefloodEnumRollup;
+import com.rackspacecloud.blueflood.types.Locator;
+import com.rackspacecloud.blueflood.types.Range;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * An interface describing behavior for reading/writing of Enum values from the
+ * metrics_enum and metrics_preaggregated_{granularity} column families.
+ *
+ * The metrics_enum column family is a table of:
+ * Row (key)    | Column1         | Value
+ * -------------|-----------------|-------------------
+ * metric name  | enum hash code  | enum string value
+ *
+ * The metrics_preaggregated_{granularity} column family is a table of:
+ * Row (key)    | Column1         | Value
+ * -------------|-----------------|-------------------
+ * metric name  | timestamp       | serialized BluefloodEnumRollup object
+ */
+public interface EnumReader {
```

Comment:
I was going under the heuristic that the [driver]*IO were all driver-specific implementations of a generic *IO interface.

I think it could go either way, whatever convention is most descriptive and easiest to follow.